peer.go

```diff
@@ -1363,3 +1367,51 @@ func (p *peer) PubKey() [33]byte {
 }
 
 // TODO(roasbeef): make all start/stop mutexes a CAS
+
+// createGetLastUpdate returns the handler which serve as a source of the last
+// update of the channel in a form of lnwire update message.
+func createGetLastUpdate(router *routing.ChannelRouter,
+	pubKey [33]byte, chanID lnwire.ChannelID) func() (*lnwire.ChannelUpdate,
+	error) {
+
+	return func() (*lnwire.ChannelUpdate, error) {
+		var edge1, edge2 *channeldb.ChannelEdgePolicy
+		cb := func(chanInfo *channeldb.ChannelEdgeInfo,
+			e1, e2 *channeldb.ChannelEdgePolicy) error {
+			if lnwire.NewChanIDFromOutPoint(&chanInfo.ChannelPoint) == chanID {
+				edge1 = e1
+				edge2 = e2
+			}
+			return nil
+		}
+
+		err := router.ForEachChannel(cb)
```

Comment:
This method will iterate over _all_ channels within the _entire_ graph. A quicker method would be to use the `GenPossibleOutpoints` method. Although this may in the worse case generate 65k candidates, in practice, the outpoint index value will be between 1 and 10, so this should exit rather quickly.